mobile/components/FileSelector.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { StyleSheet, Image, Text, View, TouchableOpacity } from 'react-native'
+import { ImagePicker } from 'expo'
+import { ReactNativeFile } from 'apollo-upload-client'
+
+import logo from '../assets/apollo-upload-logo.png'
+
+class FileSelector extends React.PureComponent {
+  _onPress = async () => {
+    const result = await ImagePicker.launchImageLibraryAsync()
+
+    if (!result.cancelled)
+      this.props.onSelected(
+        new ReactNativeFile({
+          name: result.uri.split('/').pop(),
+          type: 'image/png',
```

Comment:
You should probably infer the type from the filename instead of hard coding it. E.g. `mime.lookup(result.uri) || 'image/*'` works well (`mime.lookup` coming from `mime-types`)